### PR TITLE
Draft: Enable clippy::pedantic lints and correct them

### PR DESCRIPTION
**Pull request by svix-daniel**
Simply enables most clippy pedantic lints (with some exclusions for false positives or necessary deviations).

### CODE REVIEW

File: server/svix-server/src/v1/endpoints/attempt.rs [hunk 1]
```diff
@@ -105,24 +107,24 @@ async fn list_messageattempts(
...
 
     if let Some(status) = status {
-        query = query.filter(messageattempt::Column::Status.eq(status))
+        query = query.filter(messageattempt::Column::Status.eq(status));
     }
 
     Ok(Json(MessageAttemptOut::list_response(
-        query.all(db).await?.into_iter().map(|x| x.into()).collect(),
-        limit as usize,
+        query.all(db).await?.into_iter().map(Into::into).collect(),
```
Comment (tasn) [inline server/svix-server/src/v1/endpoints/attempt.rs:126]: I can see why this is preferred, but god, this looks worse. :)

File: server/svix-server/src/v1/endpoints/endpoint.rs [hunk 2]
```diff
@@ -51,16 +51,14 @@ pub fn validate_event_types_ids(
     }
 }
 
-pub fn validate_channels_endpoint(
-    channels: &EventChannelSet,
-) -> std::result::Result<(), ValidationError> {
+pub fn validate_channels(channels: &EventChannelSet) -> std::result::Result<(), ValidationError> {
```
Comment (tasn) [inline server/svix-server/src/v1/endpoints/endpoint.rs:54]: Why did formatting change all of a sudden?

File: server/svix-server/src/v1/endpoints/endpoint.rs [hunk 3]
```diff
@@ -449,11 +452,7 @@ async fn rotate_endpoint_secret(
 
         old_keys: Set(Some(ExpiringSigningKeys(
             iter::once(last_key)
-                .chain(
-                    old_keys
-                        .map(|x| x.0.into_iter())
-                        .unwrap_or_else(|| vec![].into_iter()),
-                )
+                .chain(old_keys.map_or_else(|| vec![].into_iter(), |x| x.0.into_iter()))
```
Comment (tasn) [inline server/svix-server/src/v1/endpoints/endpoint.rs:455]: Side comment: silly decision on Rust to have the map the SECOND parameter and the else the first. It's literally called "map or else", in this order...

File: server/svix-server/src/worker.rs [hunk 4]
```diff
@@ -179,7 +179,11 @@ async fn dispatch(
     };
     let attempt = match res {
         Ok(res) => {
-            let status_code = res.status().as_u16() as i16;
+            let status_code = res
+                .status()
+                .as_u16()
+                .try_into()
+                .expect("HTTP status does not fit into u16");
```
Comment (tasn) [inline server/svix-server/src/worker.rs:186]:
```suggestion
                .expect("HTTP status does not fit into i16");
```

**Review by tasn — COMMENTED** [review]
Made some comments. I'm OK with most of the changes, not with all of them. Some look very funky and I wonder whether we should turn them off (docstring, I'm looking at you). Also not a huge fan of all of the `try_into().expect("bla bla bla")`, it feels just tedious to write, especially for something we know is fine.

Also, if possible/easy, I'd much rather if we first got in all of the "easy ones" so that this PR is more manageable, as it's scary as it is. E.g. all of the missing `;`, the Display for error, the `to_owned()` -> `.clone()`, the removing of iter on arrays. Etc.

Everything contentious can go on its own PR.

File: server/svix-server/src/core/message_app.rs [hunk 5]
```diff
@@ -1,3 +1,5 @@
+// Lots of false positives in this file
```
[the remark this answers is not in the document]
Comment (svix-daniel) [reply server/svix-server/src/core/message_app.rs:1]: Leftover comment I forgot to delete.  I was ignoring the lint that was interpreting "PostgreSQL" as a struct name in doc comments, but figured it was just easier to rename to "postgres" so I could catch the real occurrences.

**Pull request closed without merging**